test/mock-server-connection-backend.js

```diff
@@ -106,9 +123,8 @@ var ServerConnectionBackend = function( $rootScope, $timeout ) {
     while ( pendingServerRequests.length > 0 ) {
       var request = pendingServerRequests.pop();
       var requestResponse = request.fn.apply( this, request.args );
-      $rootScope.$apply( function() {
-        request.defer.resolve( requestResponse );
-      } );
+      request.defer.resolve( requestResponse );
+      $timeout.flush();
```

Comment:
The mock's promise gets wrapped in an angular promise inside `ServerConnection`. The payload is forced to run within the `$timeout` therefore `$timeout.flush` is required. This is queued up automatically in the normal angular world by setTimeout